src/storage/config.rs

```diff
@@ -110,10 +113,43 @@ impl Default for Config {
 }
 
 impl Config {
+    fn validate_engine_type(&mut self) -> Result<(), Box<dyn Error>> {
+        let v1_kv_db_path =
+            config::canonicalize_sub_path(&self.data_dir, DEFAULT_ROCKSDB_SUB_DIR).unwrap();
+        let v2_tablet_path =
+            config::canonicalize_sub_path(&self.data_dir, DEFAULT_TABLET_SUB_DIR).unwrap();
+
+        let kv_data_exists = Path::new(&v1_kv_db_path).exists();
+        let v2_tablet_exists = Path::new(&v2_tablet_path).exists();
+        if kv_data_exists && v2_tablet_exists {
+            return Err("The data dir has subfolders for both RaftKv and PartitionRaftKv".into());
```

Comment:
"TiKV has data for both raft-kv and partitioned-raft-kv."

Reply:
> "TiKV has data for both raft-kv and partitioned-raft-kv."

It will be confusing by the sentence above, because people may think the data is shared by both raft-kv and partitioned-raft-kv. I'm refining it to "Both raft-kv and partition-raft-kv's data folders exist"

Reply:
I'm just copying your phrasing below, maybe "raft-kv and partitioned-raft-kv both have data" sound better.